### ISSUE TITLE
UTCTime loses precision.

### ISSUE THREAD

**Issue by vwwv** [fizruk/http-api-data#48]
`UTCTime` loses precision, this means if you `toHeader` and then `parseHeader` you'll get a different value from the original one. I found this problem as I was using `UTCTime` as a primary key (I know, not a good idea), sometimes I was sending it as a header, sometimes within an `aeson` JSON object, and in these circumstances, the same `UTCTime` would never match:
```
>>> t <- getCurrentTime
>>> t
2017-02-15 02:30:26.988123 UTC
>>> fromJSON $ toJSON t :: Result UTCTime
Success 2017-02-15 02:30:26.988123 UTC
>>> parseHeader $ toHeader t :: Either Text UTCTime
Right 2017-02-15 02:30:26 UTC
```

**Comment by fizruk** [comment]
Right, this is related to #41.

**Comment by fizruk** [comment]
Hmm, I've just merged #52 which supposedly should have fixed this issue, but it doesn't.
The problem is that we still lose precision when encoding (in `toHeader`):

```
>>> t <- getCurrentTime
>>> t
2017-04-20 19:05:34.514465 UTC
>>> toHeader t
"2017-04-20T19:05:34Z"
```

**Comment by fizruk** [comment]
Preparing a fix right now.
